tests/requests_spec.lua

```diff
@@ -551,4 +571,61 @@ describe("Lsp request", function()
       },
     })
   end)
+
+  it(
+    "should return correct response for " .. custom_methods.BatchCodeActions .. " - Remove unused",
+    function()
+      utils.open_file "src/batch_code_actions.ts"
+      utils.wait_for_lsp_initialization()
+
+      vim.wait(1000)
+
+      vim.cmd ":TSToolsRemoveUnused"
+
+      vim.wait(200)
```

Comment:
I wonder maybe for some of this waits you can utilize second param of wait which allow to break wait when something happens. It may offload a bit this tests from standing and do nothing. If it is possible ofc, if something isn't easily breakable let it be pure wait.

Reply:
honestly I have no idea on what to wait for :( the first `1000` ms wait is because before that tsserver does return empty code_edits, and second is because the code edits are not applied yet. I've tried to check the contents of the file but I couldn't get it to work unfortunately :(

Reply:
It's ok then.